include/revng/Model/model-schema.yml

```diff
@@ -99,40 +125,40 @@ definitions:
 
       A type definition can be a `struct`, a `union`, a `typedef`, an `enum` or a
       function prototype.
-    type: struct
+
+    key:
+      - ID
+      - Kind
+    abstract: true
+
     fields:
       - name: ID
+        doc: |-
+          A unique numeric identifier for this type.
         type: uint64_t
         is_guid: true
-        doc: |-
-          A unique identifier for this type.
+
       - name: Kind
+        doc: |-
+          Specifies which of the children classes this object is upcastable to.
         type: TypeDefinitionKind
+
       - name: Name
-        doc: |-
-          A user-chosen `Identifier` for this type.
+        doc: A user-chosen `Identifier` for this type.
         type: string
         optional: true
+
       - name: Comment
+        doc: |-
+          A doxygen comment attached to the *definition* of this type.
```

Comment:
I wouldn't imply we use doxygen, we don't in asseembly and possibly elsewhere.
Same elsewhere.

Reply:
Every mention of "doxygen" was dropped! Please take a look.